pkg/controller/storagecluster/cephcluster.go

```diff
@@ -335,8 +335,7 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rook.StorageClassDevi
 						Operator: corev1.NodeSelectorOpIn,
 						Values:   []string{topologyKeyValues[topologyIndex]},
 					}
-					nodeSelectorTerms := placement.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
-					nodeSelectorTerms[0].MatchExpressions = append(nodeSelectorTerms[0].MatchExpressions, nodeZoneSelector)
+					appendNodeRequirements(&placement, nodeZoneSelector)
```

Comment:
Not related to LabelSelector. Should be in different commit (probably with enhancement commit from below)

Reply:
Well, it is related: This is about bad behavior (crashing) on nil `placement.NodeAffinity` when we have set an empty LabelSelector.  This very line was segfaulting.

This new function `appendNodeRequirements` is also invoked in the changed function `getPlacement`.

I agree though, that we might want to first fix the crash with less refactoring / enhancements, if possible, and then rework the code.